libs/media-player/src/lib/audio-annotator.tsx

```diff
@@ -0,0 +1,262 @@
+import { isNullOrUndefined } from '@coscrad/validation-constraints';
+import {
+    ArrowLeft as ArrowLeftIcon,
+    ArrowRight as ArrowRightIcon,
+    Clear as ClearIcon,
+} from '@mui/icons-material/';
+import { Box, IconButton, Stack, Typography, styled } from '@mui/material';
+import { useEffect, useRef, useState } from 'react';
+import { asFormattedMediaTimecodeString } from './shared/as-formatted-media-timecode-string';
+import { AudioMIMEType } from './shared/audio-mime-type.enum';
+import { isAudioMIMEType } from './shared/is-audio-mime-type';
+import { KeyboardKey, useKeyDown } from './shared/use-key-down';
+import { TimeRangeSelectionVisual, TimeRangeVisualState } from './time-range-selection-visual';
+
+type Nullable<T> = T | null;
+
+export type MedidaPlayDirection = 'forward' | 'reverse';
+
+export enum KeyboardShortcuts {
+    play = KeyboardKey.spacebar,
+    clear = KeyboardKey.c,
+    markInPoint = KeyboardKey.i,
+    markOutPoint = KeyboardKey.o,
+    scrubForward = KeyboardKey.k,
+    scrubBackward = KeyboardKey.j,
+}
+
+const StyledAudioPlayer = styled('audio')`
+    border-radius: 20px;
+`;
+
+export type TimeRangeSelection = {
+    inPointSeconds: number;
+    outPointSeconds: number;
+};
+
+interface AudioAnnotatorProps {
+    audioUrl: string;
+    mimeType?: AudioMIMEType;
+    onTimeRangeSelected: (timeRangeSelected: Nullable<TimeRangeSelection>) => void;
+}
+
+export const AudioAnnotator = ({
+    audioUrl,
+    mimeType,
+    onTimeRangeSelected,
+}: AudioAnnotatorProps) => {
+    const audioRef = useRef<HTMLVideoElement>(null);
+
+    const [inPointSeconds, setInPointSeconds] = useState<Nullable<number>>(null);
+
+    const [outPointSeconds, setOutPointSeconds] = useState<Nullable<number>>(null);
+
+    const [isPlayable, setIsPlayable] = useState<boolean>(false);
+
+    const [timeRangeVisualState, setTimeRangeVisualState] = useState<TimeRangeVisualState>(null);
+
+    const [errorMessage, setErrorMessage] = useState<string>('');
+
+    useEffect(() => {
```

Comment:
I've centralized all the UI state in this `useEffect()`, hopefully it's readable.  I decided that it would be simpler to have the `timeRangeVisualState` determined here rather than pass `inPointSeconds` and `outPointSeconds` into the `TimeRangeSelectionVisual` component and have the same logic repeated there.  It also simplifies the `TimeRangeSelectionVisual` component state with only three options: 1) don't show either bar, 2) show the `inpoint-selected-bar`, or 3) show the `timerange-selected-bar`.

Reply:
As I previously showed you (with a full TypeScript code snippet) you can calculate the `TimeRangeSelectionVisual` (rename this `TimeRangeSelectionStatus`) from the state of the in point and out point. You don't need any effect or additional state. Just "project" off of inPointSeconds and outPointSeconds. It's that simple.

Reply:
That said, I can't find the snippet in this file. Maybe I put it in the wrong place.
```ts
const calculateTimeRangeSelectionStatus = (inPointMilliseconds: number |  null, outPointMilliseconds: number | null): TimeRangeSelectionStatus =>{
    if(isNull(inPointMilliseconds)) return `noSelection`;

    if(isNull(outPointMilliseconds)) return `inPointOnly`;

    return `fullSelection`;
}

export const AudioAnnotator = (props) =>{
   // use state up here

  // project the selection status off of the state
    const timeRangeSelectionStatus = calculateTimeRangeSelectionStatus(inPointSeconds,outPointSeconds)

  // ...


  // feed this in as props to your status bar component (I renamed this for clarity)
    <TimeRangeStatusBar timeRangeStatus={timeRangeStatus} />
}
```